testing/kuttl/e2e/exporter-password-change/04-assert.yaml

```
@@ -0,0 +1,39 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+commands:
+# Ensure that the password has been updated in the exporter and it can still access
+# Postgres. 
+# - Check that the DATA_SOURCE_PASS env var is set to `password`. Changing an env
+#   var means that the container has been restarted
+# - Check that all containers in the instance pod are ready after restart
+# - Grab the scrape_error output from exporter metrics and check that there were no scrape errors
+- script: |
+    retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5' - "$@"; }
+    contains() { bash -ceu '[[ "$1" == *"$2"* ]]' - "$@"; }
+    check_containers_ready() { bash -ceu ' \
+      echo "$1" | jq -e ".[] | select(.type==\"ContainersReady\") | .status==\"True\""' - "$@";
+    }
+    pod=$(kubectl get pods -o name -n $NAMESPACE \
+      -l postgres-operator.crunchydata.com/cluster=exporter-password-change \
+      -l postgres-operator.crunchydata.com/crunchy-postgres-exporter=true)
+    [ "$pod" = "" ] && retry "Pod not found" && exit 1
+    password=$(kubectl exec -n ${NAMESPACE} ${pod} -c exporter -- bash -c 'echo $DATA_SOURCE_PASS')
```

```suggestion
    password=$(kubectl exec -n ${NAMESPACE} ${pod} -c exporter -- bash -c 'printenv DATA_SOURCE_PASS')
```

I really have no reason for this suggestion. Maybe it's clearer to me that `printenv` is printing an env, but what else could `echo` be doing in this scenario?

---

Yeah, printenv probably would be clearer. With pass file changes it will turn into a `cat /file` so most likely won't change it

---

its using `cat` now